devito/finite_differences/derivative.py

```diff
@@ -359,8 +356,4 @@ def _eval_fd(self, expr):
         for e in self._ppsubs:
             res = res.xreplace(e)
 
-        # Step 5: Cast to EvaluatedDerivative
-        assert res.is_Add
-        res = EvalDerivative(*res.args, evaluate=False)
```

Comment:
Note for reviewers: now unnecessary as done directly inside finite_difference.py